src/Domain/Charity.php

```diff
@@ -7,6 +7,9 @@
 use Doctrine\ORM\Mapping as ORM;
 
 /**
+ * @psalm-api Work around Psalm being unhappy with the property or method being unused
```

Comment:
This doesn't feel like the most accurate way to make Psalm happy about `$tbgApprovedToClaimGiftAid`. The real reason it is declared and not used in PHP is that it's used via DQL / queries built into DQL instead.

(If we add a getter, it just switches to complaining about that being unused.)

Reply:
I'd add a suppression on the specific property, instead of at the class level:

```php
    /**
     * @psalm-suppress PossiblyUnusedProperty - used in a database query
     * @ORM\Column(type="boolean")
     * @var bool    Whether the charity's Gift Aid may NOW be claimed by the Big Give according to HMRC.
     */
    protected bool $tbgApprovedToClaimGiftAid = false;
    ```

Reply:
@bdsl Thanks, yep – that makes more sense. I guess I was hoping there would be an answer that would lean on the Doctrine plugin to make this inferable without suppressing anything. Do you know what that plugin does with Doctrine stuff in relation to psalm? I think it's readme was pretty empty if I was looking in the right place.

Reply:
I don't remember I'm afraid, I haven't used Psalm with Doctrine much, I think it's not a big issue, probably unusual that we'd have a property that we never read from PHP. Even if we don't strictly need to read something in a lot of cases we can put in a read that just checks that some entity isn't in a bad state for a given operation to happen, even if we know it wouldn't be because we wouldn't have selected it from the DB.

I guess we could add something like that in the Donation class, but I'm not sure I like the law of demeter violation and possible trigger of lazy loading:

![image](https://github.com/thebiggive/matchbot/assets/159481/91f08522-4827-43c2-85a1-a575b00467e9)

Reply:
Also for static analysis related to stuff in the DB I think we could do quite a bit more using PHPStan and https://github.com/staabm/phpstan-dba rather than only having Psalm. But given that matchbot doesn't have a lot of code changes probably not worth it any time soon.

Reply:
ah I see making the property private made it change from possiblyUnused to just unused in Psalm's eyes. Makes sense.